### ISSUE TITLE
bug: Markdown extra really slow

### ISSUE THREAD

**Issue by awilkins** [LazyVim/LazyVim#2051]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

NVIM v0.10.0-dev (nightly)

### Operating system/version

Ubuntu 20.04.3

### Describe the bug

Installing the Markdown extra results in Very Slow Performance ; even with relatively trivial Markdown files, ie, over 1 second to execute `j` and `k` (move down / up).

On investigation, this seems to be down to the `headlines` plugin.

### Steps To Reproduce

1. Install LazyVim repro
2. Restart once after install
3. Open a markdown buffer, restart again after parser install
4. Open markdown buffer again
5. Goof around for a few lines in a Markdown file
    - Create a couple of headers, paste the whole thing a few times
6. Observe how slow `j` and `k` get

---

1. Disable `headlines`
2. Reopen file, observe performance increase

### Expected Behavior

Not slow. I have a feeling something is broken because AFAICT the ~~Markdown LSP defines a bunch of conceals (like for bullet lists) that aren't showing up either.~~ ; not being able to see these characters is down to `conceallevel` defaulting to 3.

Seems like this problem is all `headlines`

### Repro

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
	vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
	vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
	spec = {
		{ "LazyVim/LazyVim", import = "lazyvim.plugins" },
		-- add any other plugins here
		{ import = "lazyvim.plugins.extras.lang.markdown" },
	},
}
require("lazy").setup(plugins, {
	root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here
```

**Comment by awilkins** [comment]
TLDR : `headlines` is broken on `kitty-0.15`, install a newer Kitty or disable the headlines plugin.

I worked this out : 

- It's fine on my Ubuntu 22.04 machine
- It's slow on my Ubuntu 20.04 machine

It's something to do with `kitty` : it's version 0.15 on the older version of the distro and 0.21 on the newer one.

Here's a screenshot comparing `kitty` 0.15 vs `kitty` 0.31

`headlines` doesn't render right on the older kitty, the background after the header isn't there, and there's a thin line instead of the lower part of the background ; I think this is being rendered using u1FB02 (`BLOCK SEXTANT-12`) (the upper one third of a block char) .. and Kitty 0.15 isn't handling this right.

```
The font chosen by the OS for the text: U+1fb02 is Face(family=DejaVu Sans Mono, style=Book, ps_name=DejaVuSansMono, path=/usr/share/fonts/truetype/dejavu/DejaVuSansMono.ttf, index=0, is_scalable=True, has_color=False, ascender=1901, descender=-483, height=2384, underline_position=-175, underline_thickness=90) but it does not actually contain glyphs for that text```

(doesn't log this on newer versions)

Every time `BLOCK SEXTANT-12` is turning up it's having a little panic about finding the char and trying to re-cache it, which means every instance of this char you render slows down every frame ; which means putting headlines in Markdown docs with the plugin enabled gets slow fast, and slower the wider you set your terminal.

A Markdown doc of much larger size with zero headers in it is still fast.

![image](https://github.com/LazyVim/LazyVim/assets/368399/d53d53a3-3eee-4807-8139-0bc011652ad5)
